Migrated Snapshot and Restore tests to use test user

cuff-links opened the pull request:
This PR is part of the meta issue https://github.com/elastic/kibana/issues/60815 to migrate functional tests to use `test_user`.

LeeDr commented:
@cuff-links could you mention any doc links you found that either have the required privs defined or need to have them defined?  I don't really know if `cluster: manage` should be required?

LeeDr commented:
@elasticmachine merge upstream

cuff-links commented:
> @cuff-links could you mention any doc links you found that either have the required privs defined or need to have them defined? I don't really know if `cluster: manage` should be required?

As far as the elasticsearch privileges go, I referenced https://www.elastic.co/guide/en/elasticsearch/reference/8.0/security-privileges.html. That's where I got the `snapshot` permission. `cluster:manage` I added because I saw that some previous tests did the same thing and I was doing some trial and error. I added that one after I got a CI failure showing that the user wasn't authorized. I don't find the docs to be  super clear on what's required.

cuff-links commented:
jenkins test this

cuff-links commented:
Per https://www.elastic.co/guide/en/elasticsearch/reference/master/snapshots-register-repository.html#snapshot-repo-prereqs, the required permissions are:

Cluster privileges: `monitor`, `manage_slm`, `cluster:admin/snapshot,` and `cluster:admin/repository`
Index privilege: all on the monitor index

LeeDr reviewed (approved):
LGTM - code review only

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 126011` or prevent reminders by adding the `backport:skip` label.

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create backports run `node scripts/backport --pr 126011` or prevent reminders by adding the `backport:skip` label.
